tests/test_sodar_update_samplesheet.py

```diff
@@ -0,0 +1,619 @@
+                "Ana_03-N1-DNA1",
+                "Ana_03-N1-DNA1-WGS1",
+            ],
+        ],
+        columns=[
+            "Analysis-ID",
+            "Family-ID",
+            "Paternal-ID",
+            "Maternal-ID",
+            "Sex",
+            "Phenotype",
+            "Individual-ID",
+            "Probe-ID",
+            "Barcode",
+            "Barcode-Name",
+            "Sample Name",
+            "Extract Name",
+            "Library Name",
+        ],
+    )
+
+
+def helper_update_UCS(arg_list, UCS):
+    parser = argparse.ArgumentParser()
+    UpdateSamplesheetCommand.setup_argparse(parser)
+    args = parser.parse_args(arg_list)
+    UCS.args = args
+
+    return UCS
+
+
+def test_gather_ISA_column_names(mock_isa_data, UCS_class_object):
+    from cubi_tk.sodar.update_samplesheet import ISA_NON_SETTABLE, REQUIRED_COLUMNS
+
+    study = mock_isa_data[1]
+    assay = mock_isa_data[2]
+
+    isa_names = UCS_class_object.gather_ISA_column_names(study, assay)
+
+    assert not all(col in isa_names for col in ISA_NON_SETTABLE)
+    assert all(col in isa_names for col in REQUIRED_COLUMNS)
+
+    isa_regex = re.compile(r"(Characteristics|Parameter Value|Comment)\[(.*?)]")
+    multi_colname_regex = re.compile(r"\.[0-9]+$")
+    for col in study.columns.tolist() + assay.columns.tolist():
+        colname_long = multi_colname_regex.sub("", col)
+        colname_short = isa_regex.sub(r"\2", colname_long)
+        if colname_long in ISA_NON_SETTABLE:
+            continue
+        assert colname_long in isa_names
+        assert colname_short in isa_names
+        assert isa_names[colname_short] == isa_names[colname_long]
+
+
+def test_parse_sampledata_args(mock_isa_data, UCS_class_object):
+    isa_names = UCS_class_object.gather_ISA_column_names(mock_isa_data[1], mock_isa_data[2])
+
+    # base mapping from default
+    arg_list = [
+        "-s",
+        "Ind_01",
+        "Probe_01",
+        "Ana_01",
+        "ATCG",
+        "A1",
+        "-s",
+        "Ind_02",
+        "Probe_02",
+        "Ana_02",
+        "ACTG",
+        "A2",
+        "-s",
+        "Ind_03",
+        "Probe_03",
+        "Ana_03",
+        "ATGC",
+        "A3",
+        "-d",
+        "MV-barcodes",
+        "123e4567-e89b-12d3-a456-426655440000",
+    ]
+    expected = {
+        "Family-ID": "Family",
+        "Analysis-ID": "Source Name",
+        "Paternal-ID": "Father",
+        "Maternal-ID": "Mother",
+        "Sex": "Sex",
+        "Phenotype": "Disease status",
+        "Individual-ID": "Individual-ID",
+        "Probe-ID": "Probe-ID",
+        "Barcode": "Barcode sequence",
+        "Barcode-Name": "Barcode name",
+    }
+    USC = helper_update_UCS(arg_list, UCS_class_object)
+    assert USC.parse_sampledata_args(isa_names) == expected
+
+    # manually defined mapping
+    arg_list = [
+        "--sample-fields",
+        "Dummy-ID=Source Name",
+        "Sample Name",
+        "Extract Name",
+        "barcode=Barcode sequence",
```

Comment:
I suspect this might behave unexpectedly depending on shell/quoting?

Reply:
Ah, nvm, `arg_list` is forwarded as is to `argparse`.

Reply:
ah sry, didn't fully read your comment, I can add something to the documentation though